ros2param/ros2param/verb/set.py

```diff
@@ -37,14 +37,29 @@ def add_arguments(self, parser, cli_name):  # noqa: D102
             'node_name', help='Name of the ROS node')
         arg.completer = NodeNameCompleter(
             include_hidden_nodes_key='include_hidden_nodes')
-        parser.add_argument(
-            '--include-hidden-nodes', action='store_true',
-            help='Consider hidden nodes as well')
+
         arg = parser.add_argument(
-            'parameter_name', help='Name of the parameter')
+            'parameters', nargs='*',
+            help='List of parameter name and value pairs i.e. "int_param 1 str_param hello_world"')
         arg.completer = ParameterNameCompleter()
+
         parser.add_argument(
-            'value', help='Value of the parameter')
+            '--include-hidden-nodes', action='store_true',
+            help='Consider hidden nodes as well')
+
+    def build_parameters(self, params):
```

Comment:
What happens here if the user runs the command like:

```
ros2 param set /node
```

(that is, they fail to provide *any* parameter names or values)?  Previously, that was an error (because the user didn't provide enough arguments), but now it will just try to set an empty set of parameters.  I think we should make that an error again.

Reply:
I think this will fail with system error 2 via parser.error in `RequireParameterPairAction`, @ihasdapie could you confirm?

Reply:
Yup!
https://github.com/ros2/ros2cli/blob/5547f87a6efc1a3b7894a07bee021e343977cf74/ros2param/ros2param/verb/set.py#L36-L37